src/oatpp/network/tcp/client/ConnectionProvider.cpp

```diff
@@ -252,7 +308,9 @@ oatpp::async::CoroutineStarterForResult<const std::shared_ptr<oatpp::data::strea
       if(res == 0 || error == WSAEISCONN) {
         return _return(std::make_shared<oatpp::network::tcp::Connection>(m_clientHandle));
       }
-      if(error == WSAEWOULDBLOCK || error == WSAEINPROGRESS) {
+      if (timedout()) {
```

Comment:
This won't work because of the following:

```cpp 
if(error == WSAEWOULDBLOCK || error == WSAEINPROGRESS) {
  return ioWait(m_clientHandle, oatpp::async::Action::IOEventType::IO_EVENT_WRITE);
}
```

This action says to the coroutine processor - wait until the socket is available for writing.
And there is no timeout here. It can potentially last forever.